src/librustdoc/core.rs

```diff
@@ -346,34 +346,38 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
     interface::run_compiler_in_existing_thread_pool(config, |compiler| {
```

Comment:
Use `interface::run_compiler_in_existing_thread_pool(config, |compiler| compiler.enter(|queries| {` here so you don't have to move stuff out of the `enter` closure which won't be allowed with some follow up changes. Do the same thing with `test.rs`.